### PR TITLE
EWB-2445: Add examples 1 through 6

### PR DESCRIPTION
**Pull request by Lax125**
Signed-off-by: Marcus Koh <marcus.koh@zepben.com>

# Description

Adds examples 1 through 6, as described in https://app.clickup.com/t/6929263/EWB-2444

# Associated tasks:

- No tasks blocking this one, nor blocked by this one.

# Checklist:

- [x] I have performed a self review of my own code.
- [x] I have added/updated unit tests for these changes, and if not I have explained why they are not necessary.
- [x] I have updated the changelog.
- [x] I have handled all new warnings generated by the compiler or IDE.

### CODE REVIEW

**Comment by Lax125** [comment]
@kgreav This is just a rough first draft, give it s skim and let me know of any high-level suggestions so I can revise it further. I combined fetching the model and inspecting it into the same example.

File: src/zepben/examples/fetching_network_model.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+from zepben.evolve import Conductor, PowerTransformer, connect_with_password, SyncNetworkConsumerClient, ConductingEquipment, EnergyConsumer, Switch, \
+    SinglePhaseKind, connected_equipment_trace, current_connected_equipment_trace, ConductingEquipmentStep
+from zepben.protobuf.nc.nc_requests_pb2 import INCLUDE_ENERGIZED_LV_FEEDERS
+
+
+def main():
+    channel = connect_with_password(host="dev.ewb.essential.zepben.com",
```
[the remark this answers is not in the document]
Comment (Lax125) [reply src/zepben/examples/fetching_network_model.py:12]: @sonatype-lift ignore

File: src/zepben/examples/network_hierarchy.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+gr_nsw = GeographicalRegion(mrid="gr_nsw", sub_geographical_regions=[sgr_sydney, sgr_newcastle])
+
+network_hierarchy = NetworkHierarchy(
+    geographical_regions=[gr_nsw],
+    sub_geographical_regions=[sgr_sydney, sgr_newcastle],
+    substations=[sub1, sub2, sub3],
+    feeders=[fdr1, fdr2, fdr3, fdr4, fdr5, fdr6],
+    circuits=[circuit_sydney],
+    loop_sydney=[loop_sydney]
+)
```
Comment (kgreav) [inline src/zepben/examples/network_hierarchy.py:31]: Be nice if we could print a tree hierarchy here of this, starting from the GeographicalRegion - maybe give them human readable names too? E.g:

```
for sgr in gr_nsw.sub_geographical_regions:
    print(f"- {sgr.name}) # add some nice formatting - doesn't have to exactly be a tree, just make the structure clear
    for sub in sgr.substations:
        print(...)
```

File: src/zepben/examples/network_service_interactions.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+#########
+""")
+# Apart from identified objects, a `NetworkService` also supports names. Each identified object has exactly one mRID, but can have any number of names.
+# Each name has a name type. In this example, we add two names of type "NMI" to the network model.
+meter1 = Meter()
+meter2 = Meter()
+
+name_type = NameType(name="NMI", description="National Meter Identifier")
+name_type.get_or_add_name("02783", line)
+name_type.get_or_add_name("53220", breaker)
```
Comment (kgreav) [inline src/zepben/examples/network_service_interactions.py:134]: use correct NMIs for this - 9 numbers at least

File: src/zepben/examples/fetching_network_model.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                                    client_id="jTLCsjEHE0bp6u4dhhGMAIjtJBuI1Sr6")   # Client ID will always be this for this instance of EWB.
+    feeder_mrid = "CPM3B3"
+    print(f"Fetching {feeder_mrid}")
+    # Note you should create a new client for each Feeder you retrieve
+    # There is also a NetworkConsumerClient that is asyncio compatible, with the same API.
+    client = SyncNetworkConsumerClient(channel=channel)
+    network = client.service
+
+    # Fetch feeder and all its LvFeeders
+    client.get_equipment_container(feeder_mrid, include_energized_containers=INCLUDE_ENERGIZED_LV_FEEDERS).throw_on_error()
```
Comment (kgreav) [inline src/zepben/examples/fetching_network_model.py:26]: can we have a few similar examples for the other include_energized_containers and include_energizing_containers functions with descriptions?

File: src/zepben/examples/connecting_to_grpc_service.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+    # Specify authentication config explicitly
+    async with connect_with_secret("client ID", "client secret", "hostname", 1234,
+                                     audience="https://evolve-ewb/", issuer_domain="zepben-dev.au.auth0.com", auth_method=AuthMethod.AUTH0) as secure_channel:
+        client = NetworkConsumerClient(secure_channel)
+        grpc_result = await client.get_network_hierarchy()
+        print(grpc_result.result)
+
+
+# You may use `SyncNetworkConsumerClient` if you prefer not to use asyncio
```
Comment (kgreav) [inline src/zepben/examples/connecting_to_grpc_service.py:62]: Maybe just add a comment here that the API on SyncNetworkConsumerClient is the same

**Review by kgreav — CHANGES_REQUESTED** [review]
Looking really good - maybe just add a really contrived network example (like source -> line -> fuse -> transformer -> fuse -> breaker -> line -> consumer) with an ASCII diagram.
In that example can you also document and use NetworkService.add_connectivity_node() 

Once you have the above maybe use it to give some examples exploring connectivity? of that model?

File: src/zepben/examples/fetching_network_model.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+from zepben.evolve import Conductor, PowerTransformer, connect_with_password, SyncNetworkConsumerClient, ConductingEquipment, EnergyConsumer, Switch, \
+    SinglePhaseKind, connected_equipment_trace, current_connected_equipment_trace, ConductingEquipmentStep
+from zepben.protobuf.nc.nc_requests_pb2 import INCLUDE_ENERGIZED_LV_FEEDERS
+
+
+def main():
+    # See connecting_to_grpc_service.py for examples of each connect function
+    channel = connect_with_password(host="EWB hostname",
```
[the remark this answers is not in the document]
Comment (Lax125) [reply src/zepben/examples/fetching_network_model.py:13]: @sonatype-lift ignore

File: src/zepben/examples/network_hierarchy.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+sub1 = Substation(normal_energized_feeders=[fdr1, fdr2])
+sub2 = Substation(normal_energized_feeders=[fdr3])
+sub3 = Substation(normal_energized_feeders=[fdr4, fdr5, fdr6])
+
+sgr_sydney = SubGeographicalRegion(mrid="sgr_sydney", substations=[sub1, sub2])
+sgr_newcastle = SubGeographicalRegion(mrid="sgr_newcastle", substations=[sub3])
+
+gr_nsw = GeographicalRegion(mrid="gr_nsw", sub_geographical_regions=[sgr_sydney, sgr_newcastle])
+
+nh = NetworkHierarchy(
```
[the remark this answers is not in the document]
Comment (Lax125) [reply src/zepben/examples/network_hierarchy.py]: @sonatype-lift ignore

File: src/zepben/examples/ieee_13_node_test_feeder.py [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+ec_611_phc = EnergyConsumerPhase(phase=SinglePhaseKind.C, p=170, q=80)
+ec_611 = EnergyConsumer(terminals=[ec_611_t], energy_consumer_phases=[ec_611_pha, ec_611_phb, ec_611_phc])
+
+# Distributed load on line 632-671 is unmodelled.
+
+######################
+# SETTING PARAMETERS #
+######################
+def plsi_from_z_per_mile(*impedances: Tuple[float, float]):
+    r_per_foot, x_per_foot = numpy.mean(impedances, axis=0) / FEET_PER_MILE
```
[the remark this answers is not in the document]
Comment (Lax125) [reply src/zepben/examples/ieee_13_node_test_feeder.py:128]: @sonatype-lift ignore

File: src/zepben/examples/network_hierarchy.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+sub3 = Substation(normal_energized_feeders=[fdr4, fdr5, fdr6])
+
+circuit_sydney = Circuit(end_substations=[sub1, sub2])
+loop_sydney = Loop(circuits=[circuit_sydney], substations=[sub1], energizing_substations=[sub2])
+sgr_sydney = SubGeographicalRegion(mrid="sgr_sydney", substations=[sub1, sub2])
+sgr_newcastle = SubGeographicalRegion(mrid="sgr_newcastle", substations=[sub3])
+
+gr_nsw = GeographicalRegion(mrid="gr_nsw", sub_geographical_regions=[sgr_sydney, sgr_newcastle])
+
+network_hierarchy = NetworkHierarchy(
```
[the remark this answers is not in the document]
Comment (Lax125) [reply src/zepben/examples/network_hierarchy.py:29]: @sonatype-lift ignore

File: src/zepben/examples/network_service_interactions.py [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+#########
+# NAMES #
+#########
+""")
+# Apart from identified objects, a `NetworkService` also supports names. Each identified object has exactly one mRID, but can have any number of names.
+# Each name has a name type. In this example, we add two names of type "NMI" to the network model.
+meter1 = Meter()
+meter2 = Meter()
+
+name_type = NameType(name="NMI", description="National Meter Identifier")
```
[the remark this answers is not in the document]
Comment (Lax125) [reply src/zepben/examples/network_service_interactions.py:132]: @sonatype-lift ignore

**Comment by Lax125** [comment]
Tests not needed for example usage code.

**Review by kgreav — APPROVED** [review]

**Pull request merged**